openapi-style-guide.md

```diff
@@ -235,6 +242,39 @@ For "oauth2" security schemes, `scopes` must contain at least one entry.
 
 The key of each entry in `scopes` must be of the form "<resource URI>/scope name", where "scope name" is typically ".default" for Azure services.
 
+#### Examples
+
+The following example shows how to define a security scheme for Azure Active Directory authentication:
+```
+  "securityDefinitions": {
+    "MySecurityScheme": {
+      "type": "oauth2",
+      "tokenUrl": "https://login.microsoftonline.com/common/oauth2/v2.0/token",
+      "flow": "application",
+      "description": "Azure Active Directory OAuth2 authentication",
+      "scopes": {
+        "https://resource.microsoft.com/.default": "Client credential scope"
+      }
+    }
+  }
+```
+
+Note that the name "MySecurityScheme" has no significance, and "https://resource.microsoft.com/" is meant to signify the URI of the public cloud resource.
+
+The "flow" is not particularly relevant but the "implicit" oauth2 flow is now considered insecure so another choice like "application" is preferable.
+
+The following example defines an apikey security scheme:
+```
+  "securityDefinitions": {
+    "MyApiKeySecurityScheme": {
```

Comment:
Same here regarding using a common name. In fact, I think we should recommend "AzureKey" because - good or bad - .NET's codegen, if not other languages' codegen, requires "AzureKey" to do the right thing, thus leading to transforms like this:

```yaml
directive:
# Support automatically generating code for key credentials.
- from: swagger-document
  where: $.securityDefinitions
  transform: |
    $["AzureKey"] = $["apim_key"];
    delete $["apim_key"];

- from: swagger-document
  where: $.security
  transform: |
    $ = [
        {
          "AzureKey": []
        }
    ];
```

Reply:
According to t[his issue](https://github.com/Azure/autorest.csharp/issues/2222), the .NET generator has been fixed to use only the "type" and not the name of the security scheme. So hopefully that transform can be eliminated an it will all "just work".

But I don't mind changing the name, and I'll add the caveat as above that the name is not significant.

Reply:
Ah, excellent! I didn't realize that. Hopefully it's not a problem elsewhere. Still seems like keeping it familiar might be *slightly* beneficial, but either way: yeah, keep the note it doesn't really matter.